internal/templates/template.go

```diff
@@ -23,16 +25,23 @@ func Init(w io.Writer, template, dir string) error {
 	}
 
 	templatePath, ok := templates[template]
+	cloneURL := "https://github.com/jetpack-io/devbox.git"
 	if !ok {
-		return usererr.New("unknown template %q", template)
+		u, err := url.Parse(template)
+		splits := strings.Split(u.Path, "/")
+		if err != nil || u.Scheme != "https" || len(splits) == 0 {
+			return usererr.New("unknown template name or format %q", template)
+		}
+		templatePath = splits[len(splits)-1]
+		cloneURL = template
 	}
 
 	tmp, err := os.MkdirTemp("", "devbox-template")
 	if err != nil {
 		return errors.WithStack(err)
 	}
 	cmd := exec.Command(
-		"git", "clone", "https://github.com/jetpack-io/devbox.git", tmp,
+		fmt.Sprintf("git clone %s %s", cloneURL, tmp),
```

Comment:
Any reason to wrap all of these in a single string? It may work, but generally you want to pass each a an arg. Keep in mind that exec.Command treats the first arg differently from the rest (e.g. it will wrap in double quotes etc). So if  cloneURL of temp have any crazy characters this may fail.

Reply:
This was a mistake, I reverted it.